zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ControllableStreamClockImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.stream.impl;
+
+import io.camunda.zeebe.stream.api.StreamClock;
+import io.camunda.zeebe.stream.api.StreamClock.ControllableStreamClock.Modification.None;
+import io.camunda.zeebe.stream.api.StreamClock.ControllableStreamClock.Modification.Offset;
+import io.camunda.zeebe.stream.api.StreamClock.ControllableStreamClock.Modification.Pin;
+import java.time.Instant;
+import java.time.InstantSource;
+import java.util.Objects;
+
+public final class ControllableStreamClockImpl implements StreamClock.ControllableStreamClock {
```

Comment:
:+1: Definitely easier to read in general than the current `ControlledActorClock`